erpnext/hr/report/monthly_attendance_sheet/monthly_attendance_sheet.py

```diff
@@ -120,137 +138,258 @@ def get_chart_data(emp_att_map, days):
 
 	return chart
 
-def add_data(employee_map, att_map, filters, holiday_map, conditions, default_holiday_list, leave_list=None):
-
+def add_data(employee_map, att_map, holiday_map, filters, default_holiday_list, leave_list=None):
 	record = []
 	emp_att_map = {}
 	for emp in employee_map:
 		emp_det = employee_map.get(emp)
 		if not emp_det or emp not in att_map:
 			continue
-
 		row = []
 		if filters.group_by:
 			row += [" "]
 		row += [emp, emp_det.employee_name]
+		emp_status_map = []
 
+		to_date = getdate(filters["to_date"])
+		from_date =getdate(filters["from_date"])
+
+		keys = get_days_columns(to_date, from_date, get_att_map_key=True)
+		status_map = get_attendance_status_abbr_map()
 		total_p = total_a = total_l = total_h = total_um= 0.0
-		emp_status_map = []
-		for day in range(filters["total_days_in_month"]):
-			status = None
-			status = att_map.get(emp).get(day + 1)
 
-			if status is None and holiday_map:
-				emp_holiday_list = emp_det.holiday_list if emp_det.holiday_list else default_holiday_list
+		for day in keys:
+			attendance_detail = att_map.get(emp).get(day)
+			emp_holiday_list = emp_det.holiday_list if emp_det.holiday_list else default_holiday_list
+			status = None
+			status = get_status(attendance_detail, holiday_map, emp_holiday_list, day)
```

Comment:
```suggestion
			status = get_status(attendance_detail, holiday_map, emp_holiday_list, day)
```
You are anyway setting the status immediately so you can remove the default assignment.